libcontainer/rootfs_init_linux.go

```diff
@@ -267,6 +267,9 @@ func (l *linuxRootfsInit) Init() error {
 
 	case rootfsIDMap:
 		rootfs := l.reqs[0].Rootfs
+		uid := l.reqs[0].Uid
```

Comment:
Which requirement are you referring to? The entering of the container's mount-ns or the unmounting/remounting of overlayfs?

We enter the container's mount ns because the ID-mapping must occur within that ns.

The unmounting/remounting of overlayfs is required because the higher level manager (e.g., Docker/containerd, etc) is not setting up the user-ns and associated ID-mappings in the overlay lower-layers of the container's rootfs. Sysbox has to do that. Therefore it has to unmount the existing overlay mount, ID-map the lower layers, and remount the overlay mount. It's not a problem in the ID-map design, but rather a workaround for the fact that the higher level container manager is not using ID-mapping yet.

Reply:
Sorry, the context for my question is written in the commit description associated to this particular change -- see below:

```
When using ID-mapping on the container's rootfs, the ID-mapping
applies to all files and dirs under the container rootfs, but
not the the container's rootfs itself (i.e., not to "/").
Therefore, operations such as "mkdir /somedir" fail with EPERM,
while operations such as "mkdir /root/somedir" work fine.

To fix this, sysbox-runc must chown the container's rootfs dir
(only the top dir, not it's subdirs) to match the container's
user-namespace user:group ID mappings.
```

I was referring to this "requirement" of needing to manually chown "/".

Reply:
Oh I see; yes this was surprising to me, but I think what's happening is that the lower layers are ID-mapped, not the rootfs dir itself (i.e., the "merged" dir). Therefore, the stuff under the rootfs dir shows up with proper ownership, but we still have to deal with the ownership of the rootfs dir itself.